internal/pkg/middleware/jwtMiddleware.go

```diff
@@ -59,9 +60,20 @@ func JwtAuthMiddleware(next http.Handler, role int) http.Handler {
 		fmt.Println("setting id: ", Id)
 		ctx := context.WithValue(req.Context(), constants.UserId, Id)
 		ctx = context.WithValue(ctx, constants.Role, Role)
+
 		req = req.WithContext(ctx)
 
 		next.ServeHTTP(rw, req)
 
 	})
 }
+
+// RequestIDMiddleware generates a request ID and adds it to the request context.
+func RequestIDMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requestID := uuid.NewString()
+		ctx := context.WithValue(r.Context(), constants.RequestID, requestID)
```

Comment:
set request ID only if not present in request already

Reply:
Peerly routes are not called from other services; therefore, the request ID will not be present.